drv/stm32h7-fmc-demo-server/src/main.rs

```diff
@@ -392,6 +459,158 @@ impl idl::InOrderFmcDemoImpl for ServerImpl {
+                if byte == 6 {
+                    address += 2;
+                }
+            }
+            3 | 7 => {
+                // Peek32 / Peek32Advance
+                let b =
+                    unsafe { core::ptr::read_volatile(address as *const u32) };
+                write_chunk(b.to_le_bytes(), &mut response)?;
+                if byte == 7 {
+                    address += 4;
+                }
+            }
+            4 | 8 => {
+                // Peek64 / Peek64Advance
+                let b =
+                    unsafe { core::ptr::read_volatile(address as *const u64) };
+                write_chunk(b.to_le_bytes(), &mut response)?;
+                if byte == 8 {
+                    address += 8;
+                }
+            }
+
+            9 | 13 => {
+                // Poke8 / Poke8Advance
+                let x = read_byte(&mut packet)?;
+                unsafe {
+                    core::ptr::write_volatile(address as *mut u8, x);
+                }
+                if byte == 13 {
+                    address += 1;
+                }
+            }
+            10 | 14 => {
+                // Poke16 / Poke16Advance
+                let x = u16::from_le_bytes(read_chunk(&mut packet)?);
+                unsafe {
+                    core::ptr::write_volatile(address as *mut u16, x);
+                }
+                if byte == 14 {
+                    address += 2;
+                }
+            }
+            11 | 15 => {
+                // Poke32 / Poke32Advance
+                let x = u32::from_le_bytes(read_chunk(&mut packet)?);
+                unsafe {
+                    core::ptr::write_volatile(address as *mut u32, x);
+                }
+                if byte == 15 {
+                    address += 4;
+                }
+            }
+            12 | 16 => {
+                // Poke64 / Poke64Advance
+                let x = u64::from_le_bytes(read_chunk(&mut packet)?);
+                unsafe {
+                    core::ptr::write_volatile(address as *mut u64, x);
+                }
+                if byte == 16 {
+                    address += 8;
+                }
+            }
+            _ => return Err(NetworkError::NotUnderstood),
+        }
+    }
+
+    Ok(orig_response_len - response.len())
+}
+
+fn read_byte(buf: &mut &[u8]) -> Result<u8, NetworkError> {
+    let (&first, rest) = buf.split_first().ok_or(NetworkError::Truncated)?;
+    *buf = rest;
+    Ok(first)
+}
+
+fn read_chunk<const N: usize>(
+    buf: &mut &[u8],
+) -> Result<[u8; N], NetworkError> {
+    let (&first, rest) =
+        buf.split_first_chunk().ok_or(NetworkError::Truncated)?;
+    *buf = rest;
+    Ok(first)
+}
+
+fn write_byte(byte: u8, buf: &mut &mut [u8]) -> Result<(), NetworkError> {
+    let tbuf = take(buf);
+    let (first, rest) =
+        tbuf.split_first_mut().ok_or(NetworkError::ResponseTooBig)?;
+    *first = byte;
+    *buf = rest;
+    Ok(())
```

Comment:
What about delegating to `write_chunk`, e.g.
```suggestion
    write_chunk([byte], buf)
```

Reply:
Huh, why not